diff --git a/processes/wakeupfd.go b/processes/wakeupfd.go
index ed0f154..f902749 100644
--- a/src/github.com/armPelionEdge/maestro/processes/wakeupfd.go
+++ b/src/github.com/armPelionEdge/maestro/processes/wakeupfd.go
@@ -77,7 +77,7 @@ func (this *WakeupFd) ReadWakeup() (ret uint64, errno error) {
 // eventfd() is Linux specific
 // C def: int eventfd(unsigned int initval, int flags);
 func eventfd(initval uint, flags int) (fd int, errno error) {
-	val, _, err := syscall.RawSyscall(syscall.SYS_EVENTFD,
+	val, _, err := syscall.RawSyscall(syscall.SYS_EVENTFD2,
 		uintptr(initval),
 		uintptr(flags),
 		0)
