From 83f3e0254f9a61d109420d23544c23f31f5fe765 Mon Sep 17 00:00:00 2001
From: Yash Goyal <yash.goyal@arm.com>
Date: Sun, 3 May 2020 02:36:01 -0500
Subject: [PATCH 3/3] fix CMake test Build Type Release

---
 cmake/examples_configure.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/examples_configure.cmake b/cmake/examples_configure.cmake
index 5105e77..5e4d3f9 100644
--- a/cmake/examples_configure.cmake
+++ b/cmake/examples_configure.cmake
@@ -14,7 +14,7 @@ if (DEFINED EDGE_CLIENT_REQUEST_TIMEOUT_THRESHOLD_MS)
 endif ()
 SET (EDGE_CLIENT_REQUEST_TIMEOUT_THRESHOLD_MS 60000) # one minute
 
-if (NOT (CMAKE_BUILD_TYPE STREQUAL release))
+if (NOT (CMAKE_BUILD_TYPE STREQUAL Release))
     SET (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O0")
     SET (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O0")
 endif ()
-- 
2.24.0

