From 52467017aa9d0e173254fd251611f573d1540a56 Mon Sep 17 00:00:00 2001
From: Ari Parkkila <ari.parkkila@pelion.com>
Date: Mon, 25 Jan 2021 08:38:06 +0200
Subject: [PATCH] Disable 1V8 on usdhc2 for SDMUX


diff --git a/arch/arm/dts/imx8mm-evk-u-boot.dtsi b/arch/arm/dts/imx8mm-evk-u-boot.dtsi
index 5bc030db9d..c312d4f195 100644
--- a/arch/arm/dts/imx8mm-evk-u-boot.dtsi
+++ b/arch/arm/dts/imx8mm-evk-u-boot.dtsi
@@ -123,6 +123,7 @@
 };
 
 &usdhc2 {
+	no-1-8-v;
 	u-boot,dm-spl;
 	sd-uhs-sdr104;
 	sd-uhs-ddr50;
@@ -181,4 +182,4 @@
 	/delete-property/ assigned-clocks;
 	/delete-property/ assigned-clock-parents;
 	/delete-property/ assigned-clock-rates;
-};
\ No newline at end of file
+};
diff --git a/arch/arm/dts/imx8mm-evk.dts b/arch/arm/dts/imx8mm-evk.dts
index 09505c5ebd..7c7a7e4056 100644
--- a/arch/arm/dts/imx8mm-evk.dts
+++ b/arch/arm/dts/imx8mm-evk.dts
@@ -367,6 +367,7 @@
 };
 
 &usdhc2 {
+	no-1-8-v;
 	pinctrl-names = "default", "state_100mhz", "state_200mhz";
 	pinctrl-0 = <&pinctrl_usdhc2>, <&pinctrl_usdhc2_gpio>;
 	pinctrl-1 = <&pinctrl_usdhc2_100mhz>, <&pinctrl_usdhc2_gpio>;
-- 
2.17.1

