diff --git a/arch/arm64/boot/dts/freescale/imx8mm-evk-root.dts b/arch/arm64/boot/dts/freescale/imx8mm-evk-root.dts
index aaee64b..8e900b7 100644
--- a/arch/arm64/boot/dts/freescale/imx8mm-evk-root.dts
+++ b/arch/arm64/boot/dts/freescale/imx8mm-evk-root.dts
@@ -85,6 +85,7 @@
 };
 
 &usdhc2 {
+	no-1-8-v;
 	/* sdhc3 is used by 2nd linux, configure the pin */
 	pinctrl-0 = <&pinctrl_usdhc3>, <&pinctrl_usdhc2>, <&pinctrl_usdhc2_gpio>;
 	pinctrl-1 = <&pinctrl_usdhc3>, <&pinctrl_usdhc2_100mhz>, <&pinctrl_usdhc2_gpio>;
diff --git a/arch/arm64/boot/dts/freescale/imx8mm-evkb.dts b/arch/arm64/boot/dts/freescale/imx8mm-evkb.dts
index d767cc4..4991463 100755
--- a/arch/arm64/boot/dts/freescale/imx8mm-evkb.dts
+++ b/arch/arm64/boot/dts/freescale/imx8mm-evkb.dts
@@ -406,6 +406,7 @@
 };
 
 &usdhc2 {
+	no-1-8-v;
 	pinctrl-names = "default", "state_100mhz", "state_200mhz";
 	pinctrl-0 = <&pinctrl_usdhc2>, <&pinctrl_usdhc2_gpio>;
 	pinctrl-1 = <&pinctrl_usdhc2_100mhz>, <&pinctrl_usdhc2_gpio>;
diff --git a/arch/arm64/boot/dts/freescale/imx8mm.dtsi b/arch/arm64/boot/dts/freescale/imx8mm.dtsi
index ac0de23..f410ae1 100755
--- a/arch/arm64/boot/dts/freescale/imx8mm.dtsi
+++ b/arch/arm64/boot/dts/freescale/imx8mm.dtsi
@@ -1025,6 +1025,7 @@
 			};
 
 			usdhc2: mmc@30b50000 {
+				no-1-8-v;
 				compatible = "fsl,imx8mm-usdhc", "fsl,imx7d-usdhc";
 				reg = <0x30b50000 0x10000>;
 				interrupts = <GIC_SPI 23 IRQ_TYPE_LEVEL_HIGH>;
