From 536709ded5ff28f82c42bdc04a1378255a3a0371 Mon Sep 17 00:00:00 2001
From: Ari Parkkila <ari.parkkila@pelion.com>
Date: Tue, 30 Mar 2021 11:10:45 +0300
Subject: [PATCH] Read EMAC from EEPROM on uz3eg-iocc


diff --git a/meta-lmp-bsp/dynamic-layers/xilinx/recipes-bsp/device-tree/device-tree/uz3eg-iocc/system-board.dtsi b/meta-lmp-bsp/dynamic-layers/xilinx/recipes-bsp/device-tree/device-tree/uz3eg-iocc/system-board.dtsi
index 1750937..7b01a4b 100644
--- a/meta-lmp-bsp/dynamic-layers/xilinx/recipes-bsp/device-tree/device-tree/uz3eg-iocc/system-board.dtsi
+++ b/meta-lmp-bsp/dynamic-layers/xilinx/recipes-bsp/device-tree/device-tree/uz3eg-iocc/system-board.dtsi
@@ -30,6 +30,11 @@
 			linux,default-trigger = "heartbeat";
 		};
 	};
+
+        chosen {
+                xlnx,eeprom = &eeprom; /* EMAC on EEPROM */
+        };
+
 };
 
 /* Ethernet0 with hard-coded MAC */
@@ -37,7 +42,6 @@
 	status = "okay";
 	phy-handle = <&phy0>;
 	phy-mode = "rgmii-id";
-	local-mac-address = [00 0a 35 00 02 90];
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_gem3_default>;
 	phy0: ethernet-phy@9 {
@@ -69,7 +73,7 @@
 			#size-cells = <0>;
 			reg = <0>;
 			/* IIC_EEPROM */
-			eeprom@51 { /* U5 on UZ3EG IOCC and U7 on the UZ7EV EVCC*/
+			eeprom: eeprom@51 { /* U5 on UZ3EG IOCC and U7 on the UZ7EV EVCC*/
 				compatible = "atmel,24c08";
 				reg = <0x51>;
 				#address-cells = <0x01>;
-- 
2.17.1

